sentry/src/main/java/io/sentry/clientreport/AtomicClientReportStorage.java

```diff
@@ -0,0 +1,80 @@
+package io.sentry.clientreport;
+
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.transport.DataCategory;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class AtomicClientReportStorage implements ClientReportStorage {
+
+  private final @NotNull Map<ClientReportKey, AtomicLong> lostEventCounts;
+
+  public AtomicClientReportStorage() {
+    final Map<ClientReportKey, AtomicLong> modifyableEventCountsForInit = new ConcurrentHashMap<>();
```

Comment:
`l`: As we are not modifying the Map, I think we could even use a `HashMap` instead, which doesn't require any synchronization and should be faster.

Reply:
Couldn't see a significant difference in my tests so I kept the concurrent version but can replace it with `HashMap`.

Reply:
It could be because we only do simple lookups and never write to it. You can also keep it.